src/daemon/filesystem.cpp

```diff
@@ -54,22 +58,51 @@ void Filesystem::scan(QString dirPath, bool recursive)
         }
 
         if(!entry.isDir()) {
-            emit fileFound(path);
+            emit fileFound(location.localDir, location.name, path);
             m_existingFiles.insert(path);
         } else if (entry.isDir() && !m_watcher.directories().contains(path)){
             m_watcher.addPath(path);
+            m_watcherLocations[dirPath]=location;
         }
     }
 }
 
 void Filesystem::rescan()
 {
-    QFileInfo dirInfo(m_localPath);
-    if (!dirInfo.exists() || !dirInfo.isDir()) {
-        qWarning() << "invalid path for watching" << dirInfo.absolutePath();
+    QList<WatchedLocation> pictureLocations;
+
+    QString internalPictureStorage = QStandardPaths::writableLocation(QStandardPaths::PicturesLocation);
+    if (QFileInfo(internalPictureStorage).exists()) {
+        pictureLocations << WatchedLocation{internalPictureStorage, "Internal"};
+    }
+
+    // iterate through mount volumes and try to find external drives with Pictures
+    // TODO: use card UUID and some kind of persistent configuration to proper support of multiple SD cards
+    int sdCardNum = 0;
+    for (const QStorageInfo &storage : QStorageInfo::mountedVolumes()){
+        QString mountPoint = storage.rootPath();
+        QString pictureDir = mountPoint + QDir::separator() + "Pictures";
+
+        if (storage.isValid() &&
+            storage.isReady() &&
+            mountPoint.startsWith("/run/media/") && // Sailfish OS specific mount point base for SD cards!
+            QDir(pictureDir).exists()){
+
+            qDebug() << "Found photo storage:" << pictureDir;
+            pictureLocations << WatchedLocation{pictureDir, "SD Card" + (sdCardNum == 0 ? QString(): " " + QString::number(sdCardNum))};
```

Comment:
Could you please make this line shorter?

Reply:
of course